ew/cmd/slimeoid/slimeoidcmds.py

```diff
@@ -1056,236 +1316,71 @@ async def undress_slimeoid(cmd):
-
-
-async def negaslimeoidbattle(cmd):
-    if not ewutils.channel_name_is_poi(cmd.message.channel.name):
-        response = "You must go into the city to challenge an eldritch abomination."
-        return await fe_utils.send_message(cmd.client, cmd.message.channel, fe_utils.formatMessage(cmd.message.author, response))
-
-    if cmd.mentions_count > 0:
-        # Can't mention any players
-        response = "Negaslimeoids obey no masters. You'll have to address the beast directly."
+    slimeoid = EwSlimeoid(member=cmd.message.author)
+    outofspace = False
+    
+    # Lifestate check as well as establishing slimeoidtype
+    if user_data.life_state != ewcfg.life_state_corpse:
+        slimeoidtype = "Slimeoid"
+    elif user_data.life_state == ewcfg.life_state_corpse and slimeoid.sltype == ewcfg.sltype_nega:
+        slimeoidtype = "Negaslimeoid"
+    else:
+        response = "Slimeoids don't fuck with ghosts."
         return await fe_utils.send_message(cmd.client, cmd.message.channel, fe_utils.formatMessage(cmd.message.author, response))
 
-    if cmd.tokens_count < 2:
-        response = "Name the horror you wish to face."
+    # Gotta have a response
+    if cmd.tokens_count == 1:
+        response = "Hmm? Tag your {}? Uhh, spraying your slimeoid with paint sounds like abuse, and that's a different command. Specify something to put on your dog tag.".format(slimeoidtype)
         return await fe_utils.send_message(cmd.client, cmd.message.channel, fe_utils.formatMessage(cmd.message.author, response))
 
-    slimeoid_search = cmd.message.content[len(cmd.tokens[0]):].lower().strip()
-
-    author = cmd.message.author
-
-    challenger = EwUser(member=author)
-    challenger_slimeoid = EwSlimeoid(member=author)
-
-    # Player has to be alive
-    if challenger.life_state == ewcfg.life_state_corpse:
-        response = "Your Slimeoid won't battle for you while you're dead.".format(author.display_name).replace("@", "\{at\}")
-        return await fe_utils.send_message(cmd.client, cmd.message.channel, fe_utils.formatMessage(author, response))
-
-    potential_challengees = ewutils.get_slimeoids_in_poi(id_server=cmd.guild.id, poi=challenger.poi, sltype=ewcfg.sltype_nega)
-
-    challengee_slimeoid = None
-    for id_slimeoid in potential_challengees:
-
-        slimeoid_data = EwSlimeoid(id_slimeoid=id_slimeoid)
-        name = slimeoid_data.name.lower()
-        if slimeoid_search in name:
-            challengee_slimeoid = slimeoid_data
-            break
-
-    if challengee_slimeoid is None:
-        response = "There is no Negaslimeoid by that name here."
-        return await fe_utils.send_message(cmd.client, cmd.message.channel, fe_utils.formatMessage(cmd.message.author, response))
+    # If the player doesn't have a dog tag, they can't tag their slimeoid.
+    dogtag = bknd_item.find_item(item_search='dogtag', id_user=cmd.message.author.id, id_server=cmd.guild.id if cmd.guild is not None else None, item_type_filter=ewcfg.it_cosmetic)
```

Comment:
if you do it this way, any item with the name 'dogtag' can be used as a dog tag here. doing a check using the template value would fix this